fair_mango/metrics/base.py

```diff
@@ -213,10 +213,9 @@ def __init__(
     def __call__(self): ...
 
 
-def calculate_disparity(
+def disparity_generator(
     result_per_groups: np.ndarray, method: Literal["difference", "ratio"]
-) -> dict[tuple, np.ndarray[float]]:
-    result = {}
+) -> Generator[tuple[tuple, np.ndarray[float]], None, None]:
```

Comment:
This is one way to type it.

As you can see the generator type has 3 arguments (I must admit I forgot the deal about the last one, I'm gonna look it up). Because we so often only use `yield` for generators, we can type them using a simple `Iterator` type.

Reply:
Thank you for the detailed explanation!